src/SDiagonal.jl

```diff
@@ -17,6 +17,11 @@ SDiagonal(a::StaticMatrix{N,N,T}) where {N,T} = Diagonal(diag(a))
 size(::Type{<:SDiagonal{N}}) where {N} = (N,N)
 size(::Type{<:SDiagonal{N}}, d::Int) where {N} = d > 2 ? 1 : N
 
+Base.axes(D::SDiagonal) = (ax = axes(diag(D), 1); (ax, ax))
```

Comment:
This looks fine, for me constant propagation handles this well.

Reply:
This seems to be available on master already, so it's unnecessary here